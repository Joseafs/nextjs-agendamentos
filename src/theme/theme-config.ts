// import PoppinsTtf from '../fonts/poppins/Poppins-Regular.ttf';
import { css } from 'styled-components';
import { PropsTheme } from '~/types/theme';

export const theme: PropsTheme = {
  name: 'default',
  breakpoints: {
    keys: ['xs', 'sm', 'md', 'lg', 'xl'],
    width: {
      desktop: '1120px',
      tablet: '740px',
      mobile: '360px'
    },
    values: {
      xs: 512,
      sm: 768,
      md: 1024,
      lg: 1280,
      xl: 1536
    }
  },
  palette: {
    mode: 'light',
    common: {
      black: '#000',
      white: '#fff'
    },
    primary: {
      main: '#0033CC',
      light: '#3DA8F5',
      dark: '#002699',
      text: '#fff'
    },
    secondary: {
      main: '#0FAC0F'
    },
    error: {
      main: '#C20A0A',
      light: '#F76D6D',
      dark: '#AA0909',
      text: '#FEE7E7'
    },
    warning: {
      main: '#FEF6E7'
    },
    info: {
      main: '#E5ECFF'
    },
    success: {
      main: '#08916F'
    },
    grey: {
      50: '#fafafa',
      100: '#f5f5f5',
      200: '#eeeeee',
      300: '#e0e0e0',
      400: '#bdbdbd',
      500: '#9e9e9e',
      600: '#757575',
      700: '#666666',
      800: '#495057',
      900: '#333333',
      1000: '#111'
    },
    text: {
      primary: 'rgba(0,0,0,.87)',
      secondary: 'rgba(0,0,0,.6)',
      disabled: 'rgba(0,0,0,.38)'
    },
    background: {
      default: '#fff',
      divider: 'rgba(0,0,0,.1)',
      rgba: 'rgba(255,255,255,.7)'
    }
  },
  shape: {
    radius: 4
  },
  space: 8,
  shadows: [
    'none',
    '0 4px 8px rgba(0, 0, 0, 0.1)',
    '0 -4px 8px 0 rgba(0, 0, 0, 0.08)',
    '0 0 6px 0px rgba(0, 0, 0, .4)',
    '0 6px 8px rgba(0, 0, 0, 0.2)'
  ],
  zindex: {
    screen: 1000,
    appBar: 1100,
    modal: 1300,
    snackbar: 1400,
    tooltip: 1500,
    menu: 2000
  },
  animation: {
    transition: [
      'transition: all 0.2s ease-in-out',
      'transition: all 0.4s ease-in-out'
    ],
    hover: [
      css`
        background-color: ${({ theme }) => theme.palette.opacity.light};
      `,
      css`
        background-color: rgba(0, 0, 0, 0.2);
        box-shadow: 0 8px 16px rgba(0, 0, 0, 0.2);
        transform: translateY(-4px);
      `,
      css`
        box-shadow: 0 8px 16px rgba(0, 0, 0, 0.2);
        transform: translateY(-4px);
      `,
      css`
        background-color: rgba(0, 0, 0, 0.1);
        box-shadow: 0 8px 16px rgba(0, 0, 0, 0.1);
        transform: translateY(-4px);
      `
    ],
    active: [
      'transform: scale(0.94); transition: unset;',
      'transform: scale(0.99); transition: unset;'
    ]
  }
};

export default theme;
