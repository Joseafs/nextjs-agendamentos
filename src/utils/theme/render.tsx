const isProd = process.env.NODE_ENV === 'production';

const rootPath = isProd ? process.env.NEXT_PUBLIC_DOMAIN : '/';

export const fixPath = (path = ''): string => `${rootPath}${path}`;
