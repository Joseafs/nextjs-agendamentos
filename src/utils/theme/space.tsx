import { useTheme } from 'styled-components';

export const spaceCalc = (space: number[], type: 'margin' | 'padding') => {
  const theme = useTheme();

  const calc = (value: number) => value * theme.space;

  return space.length === 1
    ? `${type}: ${calc(space[0])}px`
    : space.length === 2
    ? `${type}: ${calc(space[0])}px ${calc(space[1])}px`
    : space.length === 3
    ? `${type}: ${calc(space[0])}px ${calc(space[1])}px ${calc(space[2])}px`
    : space.length === 4
    ? `${type}: ${calc(space[0])}px ${calc(space[1])}px ${calc(
        space[2]
      )}px ${calc(space[3])}px`
    : ``;
};

export const sizeImg = (space: string[]): {} => {
  return space.length === 1
    ? { width: space[0] }
    : space.length === 2
    ? { width: space[0], height: space[1] }
    : {};
};
