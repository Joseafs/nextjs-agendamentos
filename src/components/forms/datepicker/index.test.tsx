import { fireEvent, render } from '@testing-library/react';
import { ThemeUI } from '~/theme/theme-provider';
import { Datepicker } from '.';

const mockFn = jest.fn();

const component = (
  <ThemeUI>
    <Datepicker date={''} setDate={mockFn} label="Test Field" />
  </ThemeUI>
);

describe('Datepicker', () => {
  it('Should match snapshot', () => {
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });

  it('Should mockFn on input change', () => {
    const { queryByPlaceholderText } = render(component);
    const input = queryByPlaceholderText('00/00/0000') as HTMLInputElement;
    fireEvent.change(input, { target: { value: '10/10/2020' } });
    expect(mockFn).toBeCalled();
  });
});
