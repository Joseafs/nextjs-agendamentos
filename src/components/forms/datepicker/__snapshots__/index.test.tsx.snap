// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Datepicker Should match snapshot 1`] = `
<div>
  <div
    class="sc-5c4378d1-1 gfwTPz"
    data-testid="form--datepicker"
    value=""
  >
    <label
      class="sc-5c4378d1-0 edFNzs"
      for="form--datepicker"
    >
      Test Field
    </label>
    <div
      class="rmdp-container "
      style="display: inline-block;"
    >
      <input
        autocomplete="off"
        class="rmdp-input"
        id="form--datepicker"
        inputmode="none"
        placeholder="00/00/0000"
        type="text"
        value=""
      />
      <div
        class=""
        style="position: absolute; left: 0px; top: 0px; will-change: transform; visibility: hidden; z-index: 100;"
      >
        <div />
      </div>
    </div>
  </div>
</div>
`;
